File: gfg-Mathematics/all-divisors.js
// function printDivisors(number) {
//     for(let i = 0;i<=number;i++) {
//         if(number % i == 0) {
//             console.log(i)
//         }
//     }
// }



//efficient way 
function printDivisors(number) {
    for(let i = 0;i*i<= number;i++) {
        if(number%i==0) {
            console.log(i)
            if(i!= Math.floor(number/i)) {
                console.log(Math.floor(number/i))
            }
        }
    }
}

let n = 25

printDivisors(n)

File: gfg-Mathematics/all-divisors.cpp
#include <iostream>
using namespace std;

void printDivisors(int n) {
    for(int i = 1;i<=n;i++) {
        if(n%i == 0) {
            cout << i << endl;
        }
    }
}

int main()
{
    int number = 12; // 1 , 2 , 3, 4 , 6 and 12
    // 15 -> 1 , 3 , 5 , 15
    printDivisors(number);
}